### ISSUE TITLE
Overlay disappears when going back to 1st step

### ISSUE THREAD

**Issue by billyjacoby** [gilbarbara/react-joyride#789]
## 🐛 Bug Report

When using a multi-step tour if you travel back to the first step the overlay on the page disappears. It reappears when you go back to any step after 1, but always disappears at step 1 again.

## To Reproduce

I have created a repo that displays this behavior on a fresh CRA build [here](https://github.com/billyjacoby/react-joyride-issue)

## Expected behavior

The overlay should persist when traveling back to step number 1.

## Link to repl or repo (highly encouraged)

[codesandbox](https://codesandbox.io/s/suspicious-framework-u740kv?fontsize=14&hidenavigation=1&theme=dark)

[repo](https://github.com/billyjacoby/react-joyride-issue)

## Run `npx envinfo --system --binaries --npmPackages react-floater`

Paste the results here:

```bash
  System:
    OS: macOS 12.3
    CPU: (10) arm64 Apple M1 Pro
    Memory: 101.02 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 17.6.0 - ~/.nvm/versions/node/v17.6.0/bin/node
    Yarn: 1.22.18 - ~/.nvm/versions/node/v17.6.0/bin/yarn
    npm: 8.5.1 - ~/.nvm/versions/node/v17.6.0/bin/npm
```

**Comment by billyjacoby** [comment]
I wasn't able to figure out how to fix this via react-joyride but I was able to implement a (hacky) workaround in case anyone else runs into this issue.

I ended up adding a function to the `joyrideCallback` function that appends a node to the body if the step index is 1 and action is previous. Code snippet here:


```js
  function joyrideCallback(data) {
    // This is a hack to get around the issue of the overlay disappearing when you go back to the first step
    const { index, action } = data;
    if (index === 1 && action === "prev") {
      console.log("back to first step");
      // create the fallback overlay here, and remove it when we are not here
      let overlay = document.createElement("div");
      overlay.class = "joyride-overlay-custom";
      overlay.id = "custom-overlay";
      overlay.style.cssText =
        "cursor: pointer; height: 150vh; width: 100vw; pointer-events: auto; inset: 0px; overflow: hidden; position: absolute; z-index: 999; background-color: rgba(0, 0, 0, 0.7) !important; mix-blend-mode: hard-light;";
      document.body.appendChild(overlay);
    } else {
      if (document.getElementById("custom-overlay")) {
        document.getElementById("custom-overlay").remove();
      }
    }
  }
```

**Comment by jestrickler** [comment]
Ours starts with missing overlay on step-0, not just going back.

**Comment by jestrickler** [comment]
I noticed in the chrome devtools react components tab that the lifecycle prop on the JoyrideStep component is 'init' when the overlay is NOT displayed and changes to 'tooltip' when I click next and the overlay is displayed. If I manually update it in devtools to 'tooltip' on the 1st step, it displays the overlay.

**Comment by jestrickler** [comment]
This seemed to have to do with timing. I was calling the helpers.reset(true) when run is true in my own useEffect. I moved this to the callback on ACTIONS.CLOSE and I always have an overlay (knock on wood).

**Comment by gs96222** [comment]
If you are using control mode, I am just doing a restart of the Tour, which seems to be a good hacky workaround for me.

**Comment by terragady** [comment]
how do you do restart of the tour? I am having this issue of  dropping the overlay and not able to click next on the 1st step when

**Comment by jestrickler** [comment]
I have state passed from the parent component where I use a menu item to start the tour:
`const [runTour, setRunTour] = useState(false)`

Then I use the joyride helpers and callback to reset the tour:
```
const [joyride, setJoyride] = useState()

const handleCallback = ({ action }) => {
  if (ACTIONS.CLOSE === action) {
    setRunTour(false)
    joyride.reset(true)
  }
}

<Joyride callback={handleCallback} getHelpers={setJoyride} run={runTour} ... />
```

**Comment by gilbarbara** [comment]
Fixed in 2.5.2
